packages/react-native-executorch/common/rnexecutorch/models/ocr/Detector.cpp

```diff
@@ -0,0 +1,89 @@
+#include "Detector.h"
+#include <rnexecutorch/data_processing/ImageProcessing.h>
+#include <rnexecutorch/models/ocr/Constants.h>
+#include <rnexecutorch/models/ocr/DetectorUtils.h>
+
+namespace rnexecutorch {
+
+/*
+The model used as detector is based on CRAFT (Character Region Awareness for
+Text Detection) paper. https://arxiv.org/pdf/1904.01941
+*/
+
+Detector::Detector(const std::string &modelSource,
+                   std::shared_ptr<react::CallInvoker> callInvoker)
+    : BaseModel(modelSource, callInvoker) {
+  auto inputShapes = getAllInputShapes();
+  if (inputShapes.empty()) {
+    throw std::runtime_error(
+        "Detector model seems to not take any input tensors.");
+  }
+  std::vector<int32_t> modelInputShape = inputShapes[0];
+  if (modelInputShape.size() < 2) {
+    throw std::runtime_error("Unexpected detector model input size, expected "
+                             "at least 2 dimensions but got: " +
+                             std::to_string(modelInputShape.size()) + ".");
+  }
+  modelImageSize = cv::Size(modelInputShape[modelInputShape.size() - 1],
+                            modelInputShape[modelInputShape.size() - 2]);
+}
+
+cv::Size Detector::getModelImageSize() const noexcept { return modelImageSize; }
+
+std::vector<DetectorBBox> Detector::generate(const cv::Mat &inputImage) {
+  /*
+   Detector as an input accepts tensor with a shape of [1, 3, 800, 800].
+   Due to big influence of resize to quality of recognition the image preserves
+   original aspect ratio and the missing parts are filled with padding.
+   */
+  auto inputShapes = getAllInputShapes();
+  cv::Mat resizedInputImage =
+      imageprocessing::resizePadded(inputImage, getModelImageSize());
+  TensorPtr inputTensor = imageprocessing::getTensorFromMatrix(
+      inputShapes[0], resizedInputImage, ocr::mean, ocr::variance);
+
+  auto forwardResult = BaseModel::forward(inputTensor);
+  if (!forwardResult.ok()) {
+    throw std::runtime_error(
+        "Failed to forward, error: " +
+        std::to_string(static_cast<uint32_t>(forwardResult.error())));
+  }
+
+  return postprocess(forwardResult->at(0).toTensor());
+}
+
+std::vector<DetectorBBox>
+Detector::postprocess(const Tensor &tensor) const noexcept {
```

Comment:
I think that noexcept here is too aggressive since ocr::getDetBoxesFromTextMap has a CV_ASSERT which my throw
```suggestion
Detector::postprocess(const Tensor &tensor) const {
```

Reply:
I sort of assumed, that our pipeline should work in such way, that if model itself did not throw, we should not get any exceptions in the post process. These assertion should be always true for our cases but probably marking it as noexcept is too aggressive as you said. Also in some cases we are using `OpenCV` functions, which are not marked as noexcept so maybe it is not safe at all. 
I've deleted all occurrences of `noexcept` in the `postprocess` functions (`Detector,` `VerticalDetector`, `Recognizer`).

Reply:
Generally, noexcept should be used if you are 100% sure that the functions won't throw. Functions from OpenCV allocate dynamic memory, operates on raw pointers etc., so there is a whole lot of potential here for unexpected errors.